app/Notifications/NewBon.php

```diff
@@ -68,7 +68,7 @@ public function toArray($notifiable)
         return [
             'title' => $this->sender.' Has Gifted You '.$this->transaction->cost.' BON',
             'body'  => $this->sender.' has gifted you '.$this->transaction->cost.' BON with the following note: '.$this->transaction->comment,
-            'url'   => "/users/{$this->transaction->senderObj->username}",
+            'url'   => sprintf('/users/%s', $this->transaction->senderObj->username),
```

Comment:
This may not be the place to address this, these URLs should be generated using calls to `route()` or similar. Many other Notification classes have these hard-coded URL segments. It might be worth creating an Issue or PR to fix this before we forget.

Reply:
agree

Reply:
Sounds good. I notice that many routes do not have names, either, so you'll have to do add those first.